### PR TITLE
fixed #16248: .wasm / .asm.js files are both downloaded on web platform

### LINKED ISSUE #16248: [bug] .wasm / .asm.js files are both downloaded on web platform

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #16248

### Changelog

* fixed #16248: .wasm / .asm.js files are both downloaded on web platform

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/3d/misc/mesh-codec.ts [hunk 1]
```diff
@@ -21,13 +21,13 @@
  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  THE SOFTWARE.
 */
-import { CULL_MESHOPT, WASM_SUPPORT_MODE } from 'internal:constants';
+import { CULL_MESHOPT, NATIVE_CODE_BUNDLE_MODE } from 'internal:constants';
```
Comment (minggo) [inline cocos/3d/misc/mesh-codec.ts:24]: I think WASM_SUPPORT_MODE is more meaningful than NATIVE_CODE_BUNDLE_MODE.

File: pal/system-info/enum-type/feature.ts [hunk 2]
```diff
@@ -107,9 +107,9 @@ export enum Feature {
      */
     EVENT_HANDHELD = 'EVENT_HANDHELD',
     /**
-     * @en Check whether Webassembly is supported at runtime. Generally, it needs to be checked when the constant `WASM_SUPPORT_MODE` is 2.
+     * @en Check whether Webassembly is supported at runtime. Generally, it needs to be checked when the constant `NATIVE_CODE_BUNDLE_MODE` is 2.
```
Comment (minggo) [inline pal/system-info/enum-type/feature.ts:110]: `Check whether Webassembly is supported at runtime` the description is not correct now.

File: cocos/3d/misc/mesh-codec.ts (hunk 1 above)
Comment (dumganhar) [reply cocos/3d/misc/mesh-codec.ts:24]: I don't think so. 
If using asmjs, it's a normal js code. asmjs is not a wasm mode.

File: pal/system-info/enum-type/feature.ts (hunk 2 above)
Comment (dumganhar) [reply pal/system-info/enum-type/feature.ts:110]: It's correct. It's wasm support feature.
This comment means: This is a value for indicating whether the browser supports WASM. User needs to check this `WASM feature` when `NATIVE_CODE_BUNDLE_MODE` is `asmjs + wasm`.

File: cocos/3d/misc/mesh-codec.ts [hunk 3]
```diff
@@ -63,31 +63,32 @@ function initDecoderWASM (wasm_factory: any, wasm_url: string): Promise<void> {
...
+                { default: meshopt_wasm_url },
+            ]) => initDecoderWASM(meshopt_wasm_factory, meshopt_wasm_url));
         } else {
-            return initDecoderASM(meshopt_asm_factory);
+            return import('external:emscripten/meshopt/meshopt_decoder.asm.js').then(
+                ({ default: meshopt_asm_factory }) => initDecoderASM(meshopt_asm_factory),
+            );
         }
-    }));
+    }).catch(errorReport);
```
Comment (minggo) [inline cocos/3d/misc/mesh-codec.ts:91]: Is it better to print the error log?

Comment (dumganhar) [reply cocos/3d/misc/mesh-codec.ts:91]: Yes, `errorReport` is a function for print error log.
```ts
 const errorReport = (msg: any): void => { error(msg); };
```
It's the old code written like this.

Comment (minggo) [reply cocos/3d/misc/mesh-codec.ts:91]: Got it, good to know.

**Comment by minggo** [comment]
Please fix eslint error as possible.

**Review by minggo — APPROVED** [review]

**Pull request merged**